src/main/java/org/dataone/hashstore/HashStoreFactory.java

```diff
@@ -0,0 +1,87 @@
+package org.dataone.hashstore;
+
+import java.io.IOException;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.util.HashMap;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.dataone.hashstore.exceptions.HashStoreFactoryException;
+
+/**
+ * HashStoreFactory is a factory class that generates HashStore, a
+ * content-addressable file management system.
+ */
+public class HashStoreFactory {
+    private static final Log logHashStore = LogFactory.getLog(HashStoreFactory.class);
+
+    /**
+     * Factory method to generate a Hashstore
+     * 
+     * @param classPackage    String of the package name, ex.
+     *                        "org.dataone.hashstore.filehashstore.FileHashStore"
+     * @param storeProperties HashMap of the HashStore required properties:
+     *                        (Path) storePath, (int) storeDepth, (int) StoreWidth,
+     *                        (int)
+     *                        storeAlgorithm
+     * 
+     * @return HashStore instance ready to store objects and metadata
+     * @throws HashStoreFactoryException When HashStore failÏs to initialize due to
+     *                                   permissions or class-related issues
+     * @throws IOException               When there is an issue with properties
+     */
+    public static HashStore getHashStore(String classPackage, HashMap<String, Object> storeProperties)
```

Comment:
Thank you for linking the article for additional context/clarification. I have refactored HashStore/HashStoreFactory to be instantiated with a `Properties` object.

Reply:
> Sticking with convention, this would ideally be a java.io.Properties object, not a HashMap.

In general I agree with this, but I do note we are aiming for cross-language compatibility in this library, and the Python implementation and the Java implementation will share a common configuration file (literally the same file, not just an equivalent, as both will be reading from the same hashstore instance). Thus I think we decided on using YAML for the serialization syntax for the config file. I think I was the one that suggested that a simple HashMap would be a good generic structure for passing the name-value pairs from the YAML file back to the application, and would correspond nicely to the python dictionary structure, which would make for API parity across implementations. If you still prefer to use a `Properties` object here and something else in Python, then keep in mind that the serialization will not be done with the native java properties syntax, and any assumptions about properties files and property overrides would be difficult to mirror on the python side.

Reply:
@mbjones To make sure we're on the same page, I just want to clarify that I've only refactored `FileHashStore` to instantiate with a `Properties` object (to make it simpler for the calling app) - but the underlying code to get/compare `hashstore.yaml` still utilizes a HashMap. The config file is still the same as well - so we should still be good with the cross-language compatibility.

Reply:
My understanding agrees with Dou's. This props object isn't directly used by the yaml-manipulation code

Reply:
ok, sounds good